### PR TITLE
Improves #1644 - Adds integration test for /v2/queue.

### PR DESCRIPTION
**Pull request by gkleiman**

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/integration/TaskQueueIntegrationTest.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package mesosphere.marathon.integration
+
+import mesosphere.marathon.integration.setup._
+import org.scalatest.{ GivenWhenThen, Matchers }
+
+class TaskQueueIntegrationTest extends IntegrationFunSuite with SingleMarathonIntegrationTest with GivenWhenThen with Matchers {
+  test("GET /v2/queue") {
+    Given("a new app that is not healthy")
+    val appId = testBasePath / "failing"
+    val app = v2AppProxy(appId, "v1", instances = 2, withHealth = true)
```
Comment (aquamatthias) [inline src/test/scala/mesosphere/marathon/integration/TaskQueueIntegrationTest.scala]: That is not a safe way for an entry in the queue. Consider using constraints to make it impossible to launch:
```scala
val c = Protos.Constraint.newBuilder().setField("notExistent").setOperator(Operator.CLUSTER).setValue("na").build()
val app = V2AppDefinition("/app".toRootTestPath, constraints = Set(c), cmd = Some("na"), instances = 5, ports = Seq(new JInt(23)))
```

**Comment by gkleiman** [comment]
Thanks for the feedback, I have incorporated your comments.

File: src/test/scala/mesosphere/marathon/integration/setup/MarathonFacade.scala [hunk 2]
```diff
@@ -31,6 +31,10 @@ case class LeaderResult(leader: String)
 
 case class ListDeployments(deployments: Seq[Deployment])
 
+case class ITaskDelay(timeLeftSeconds: Int, overdue: Boolean)
+case class ITask(app: V2AppDefinition, count: Int, delay: ITaskDelay)
```
Comment (aquamatthias) [inline src/test/scala/mesosphere/marathon/integration/setup/MarathonFacade.scala]: The name is bad. What about TaskQueueItem?

File: src/test/scala/mesosphere/marathon/integration/setup/MarathonFacade.scala [hunk 3]
```diff
@@ -31,6 +31,10 @@ case class LeaderResult(leader: String)
 
 case class ListDeployments(deployments: Seq[Deployment])
 
+case class ITaskDelay(timeLeftSeconds: Int, overdue: Boolean)
```
Comment (aquamatthias) [inline src/test/scala/mesosphere/marathon/integration/setup/MarathonFacade.scala]: The name is bad. What about TaskQueueDelay?

**Comment by aquamatthias** [comment]
This PR needs a rebase.

**Comment by gkleiman** [comment]
Rebased!

I renamed not only my new types, but also the existing ones.

**Comment by aquamatthias** [comment]
LGTM - waiting for tests to get green.

**Pull request merged**